libs/shared/type-site/src/lib/i-style.ts

```diff
@@ -7,11 +7,16 @@ export interface IStyleEntry {
   value: string
 }
 
+export interface IStyleEntryWithInherited extends IStyleEntry {
+  inherited: boolean
+}
```

Comment:
Hi @sampullman 

I think the behavior mentioned in the issue is observed because we didn't check if the new style is originated (inherited) from a mixin when applying it on a component. 

My current solution introduced a new type `IStyleEntryWithInherited`, and use it to replace the `IStyleEntry` in `ISetComponentCustomStyleData`. If a style is inherited from somewhere else, we will not apply it to the effected component. It's not very comprehensive, but I didn't find a better way to fix this. 😕

Reply:
It's a clever solution, but it's kind of hiding the not-so-great code I wrote for tracking style changes. I think it's correct to track whether the updated style was not inherited (or similarly, if the style is "new"), but I wonder if it can be checked at a higher level than `set-component-custom-style.ts`.

Let me think about whether it's possible to do this with only modifications to `editCommands` and the style edit helper functions. If so, I'll open a separate PR that includes your other changes, otherwise I'll merge this right away!